test/powershell/Modules/Microsoft.PowerShell.Core/Get-Command.Tests.ps1

```diff
@@ -263,4 +264,8 @@ Describe "Get-Command Tests" -Tags "CI" {
         VerifyDynamicParametersExist -cmdlet $results[0] -parameterNames $paramName
         VerifyParameterType -cmdlet $results[0] -parameterName $paramName -ParameterType System.Text.Encoding
     }
+
+    It "Get-Command `" `" does not throw" -Pending {
+        Get-Command " " | Should -Not -Throw
```

Comment:
Is the expected behavior that it should not throw, or that it should throw in a *specific* way? (i.e. `CommandNotFoundException`. Get-Command "`n" throws `CommandNotFoundException`, so if a different type of whitespace results in *nothing* that would seem inconsistent?

Reply:
Guys,, do you really use whitespaces commands in your scripts and consider it a great practice?

Reply:
I stumbled upon this due to a bug in my own script; when a user enters invalid input PowerShell should respond appropriately so that the user knows what they did wrong.

`CommandNotFoundException` because space is not a valid command immediately tells you what your problem is and what you should do to fix it; `IndexOutOfRangeException` doesn't mean anything and you have to go digging through your script to find the non-obvious reason your script crashed.

While I don't have any thoughts as to whether the behavior described in #10165 should be allowed, I would present that the resolution of #10165 issue should not impact the resolution of #10148; a command either exists or it doesn't - `CommandSearcher` should be able to process any valid string without crashing in an unexpected way

Reply:
I stumbled upon this due to a bug in my own script; when a user enters invalid input PowerShell should respond appropriately so that the user knows what they did wrong.

`CommandNotFoundException` because space is not a valid command immediately tells you what your problem is and what you should do to fix it; `IndexOutOfRangeException` doesn't mean anything and you have to go digging through your script to find the non-obvious reason your script crashed. Failure to throw the right exception would also cause issues if you had an outer exception handler for `CommandNotFoundException` due to some requirement of your program

While I don't have any thoughts as to whether the behavior described in #10165 should be allowed, I would present that the resolution of #10165 issue should not impact the resolution of #10148; a command either exists or it doesn't - `CommandSearcher` should be able to process any valid string without crashing in an unexpected way

Reply:
> Guys,, do you really use whitespaces commands in your scripts and consider it a great practice?

Of course not.

That doesn't change the fact that it can be done though, and I'm still with @lordmilko: the error should be `CommandNotFoundException` if the command is not found.